### DISCUSSION TITLE
File Upload as Attachment

### DISCUSSION THREAD

**Discussion by mintdawn** [PHPMailer/PHPMailer#2561] [Q&A]
Hello, I have a form that allows users to upload a resume file. This file should be attached to the email that is sent out. How can I do this? The emails are coming through but, not the attachments. I am very new to this application and attempting to set this up for a client. So, please, be very basic in your instruction. I'd really love to learn how to make this work.

**Comment by Synchro** [comment]
Base your code on [this example](https://github.com/PHPMailer/PHPMailer/blob/master/examples/send_file_upload.phps).

**Reply by mintdawn** [reply]
Are there any variables I should be sure not to miss? its saying can not move the file and not sending.

**Reply by Synchro** [reply]
If that code is saying it can't move the file, it means that the temp directory it's trying to move it to is not writable. That's unusual (anyone should be able to write to the system temp dir), but you can point it at another directory that you do have write permissions for, for example something like:

    $uploadfile = tempnam(__DIR__ . '/uploads', hash('sha256', $_FILES['userfile']['name'])) . '.' . $ext;
